Assignment4/Fátima-Illana-d21c005/FátimaIllanaTask06.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+"""SolucionTask06.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1o7zfYeB-6579I6SGMGpznkhoMMhXliAQ
+
+**Task 06: Modifying RDF(s)**
+"""
+
+# !pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials"
+
+"""Leemos el fichero RDF de la forma que lo hemos venido haciendo"""
+
+from rdflib import Graph, Namespace, Literal
+from rdflib.namespace import RDF, RDFS
+g = Graph()
+g.namespace_manager.bind('ns', Namespace("http://somewhere#"), override=False)
+g.namespace_manager.bind('vcard', Namespace("http://www.w3.org/2001/vcard-rdf/3.0#"), override=False)
+g.parse(github_storage+"/rdf/example5.rdf", format="xml")
+
+"""Create a new class named Researcher"""
+
+ns = Namespace("http://somewhere#")
+g.add((ns.Researcher, RDF.type, RDFS.Class))
+print()
+print("GRAFO")
+print()
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.1: Create a new class named "University"**
+
+"""
+print()
+print("----------------------------------------------------------------------")
+print("TASK 6.1")
+print()
+# TO DO
+g.add((ns.University, RDF.type, RDFS.Class))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.2: Add "Researcher" as a subclass of "Person"**"""
+
+# TO DO
+print()
+print("----------------------------------------------------------------------")
+print("TASK 6.2")
+print()
+g.add((ns.Researcher, RDFS.subClassOf, ns.Person))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.3: Create a new individual of Researcher named "Jane Smith"**"""
+
+# TO DO
+print()
+print("----------------------------------------------------------------------")
+print("TASK 6.3")
+print()
+g.add((ns.JaneSmith, RDF.type, ns.Researcher))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.4: Add to the individual JaneSmith the fullName, given and family names**"""
+
+# TO DO
+print()
+print("----------------------------------------------------------------------")
+print("TASK 6.4")
+print()
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+g.add((ns.JaneSmith, VCARD.FN, Literal("Jane Smith")))
+g.add((ns.JaneSmith, VCARD.Given, Literal("Jane")))
+g.add((ns.JaneSmith, VCARD.Family, Literal("Smith")))
+# Visualize the results
+for s, p, o in g:
+  print(s,p,o)
+
+"""**TASK 6.5: Add UPM as the university where John Smith works**"""
+
+# TO DO
+print()
+print("----------------------------------------------------------------------")
+print("TASK 6.5")
+print()
+from rdflib import FOAF
+g.add((ns.UPM, RDF.type, ns.University))
+g.add((ns.JohnSmith, FOAF.workplaceHomepage, ns.UPM))
```

Comment:
The range of foaf:workplaceHomePage is a document. UPM is not a document, but a university.
Kudos for searching a property that could fit, but you need to check the domain/range to make sure they are compatible. Please find another one :)